### ISSUE TITLE
Arithmetic operator type casting documentation incorrect?

### ISSUE THREAD

**Issue by goodoldneon** [arangodb/arangodb#6106]
## My Environment

* __ArangoDB Version__:        3.3.13
* __Storage Engine__:             MMFiles / RocksDB
* __Deployment Mode__:       Single Server
* __Deployment Strategy__:   Manual Start
* __Operating System__:        Windows 10       

## Component, Query & Data

__Affected feature__:
AQL arithmetic operator type cast

__AQL query (if applicable)__:
```
let str = "3a"
return [
    0 + str == TO_NUMBER(str),
    0 + str,
    TO_NUMBER(str)
]
```

__Problem__:
Documentation states that "passing non-numeric values to an arithmetic operator will cast the operands to numbers using the type casting rules applied by the `TO_NUMBER()` function". However, `0 + "3a" == TO_NUMBER("3a")` returns `false`.

__Expected result__:
`TO_NUMBER("3a")` should equal `0 + "3a"`. However, `TO_NUMBER("3a") == 0` (as expected) but `0 + "3a" == 3` (unexpected).

**Comment by jsteemann** [comment]
Fixed in 3.4 via https://github.com/arangodb/arangodb/pull/6113

**Comment by vinaypyati** [comment]
Hi @goodoldneon,

version 3.4.0-RC1 has been released, which contains the fix for this problem. Please check it out. 
I will be closing this issue for now. Please comment, if needed, and we will reopen it.

Thanks again for reporting this issue.

Best,
